astropy/modeling/parameters.py

```diff
@@ -55,6 +56,52 @@ def _tofloat(value):
     return value
 
 
+# Helpers for implementing operator overloading on Parameter
+
+def _binary_arithmetic_operation(op, right=False):
+    @functools.wraps(op)
+    def wrapper(self, val):
+
+        if self._model is None:
+            return NotImplemented
+
+        if right:
+            return op(val, self.value)
+        else:
+            return op(self.value, val)
+
+    return wrapper
+
+
+def _binary_comparison_operation(op):
+    @functools.wraps(op)
+    def wrapper(self, val):
+
+        if self._model is None:
+            if op is operator.lt:
```

Comment:
Somewhat a matter of taste, but I would leave this in the definition of `__lt__`, so that it is easier to see the code does the correct thing.